### PR TITLE
Json mapper

### PR DESCRIPTION
**Pull request by slinkydeveloper**
Implemented a `CloudEventDataMapper` and a `CloudEventData` that uses jackson json mapper to map `CloudEventData` to pojos

### CODE REVIEW

File: formats/json-jackson/src/main/java/io/cloudevents/jackson/PojoCloudEventData.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+//            return null;
+//        }
+//        PojoCloudEventData pojoData;
+//        try {
+//            pojoData = (PojoCloudEventData) data;
+//        } catch (ClassCastException e) {
+//            throw new IllegalArgumentException("Provided event contains a CloudEventData not instance of PojoCloudEventData. Use the mapper PojoCloudEventDataMapper when you deserialize the event", e);
+//        }
+//        return (T)pojoData.getValue();
+//    }
```
Comment (aliok) [inline formats/json-jackson/src/main/java/io/cloudevents/jackson/PojoCloudEventData.java:61]: Can you get rid of these?

Comment (slinkydeveloper) [reply formats/json-jackson/src/main/java/io/cloudevents/jackson/PojoCloudEventData.java:61]: Leftover before introducing #257

**Comment by aliok** [comment]
/lgtm

Sensible follow up for https://github.com/cloudevents/sdk-java/pull/257

**Comment by johanhaleby** [comment]
I think this looks fine as well! It's in the Jackson package and people are used to deal with generics when working with Jackson.

**Review by pierDipi — APPROVED** [review]
LGTM

**Pull request merged**